### PR TITLE
added a put to the reference

### LINKED ISSUE #592: Create PUT method on Reference API
Add a PUT method to the [ReferenceAPI](https://github.com/deep-learning-indaba/Baobab/blob/bf99909f71ef7cca4923d3a93ddcea9e6d443ccf/api/app/references/api.py#L110) which updates an existing Reference using a `token` and `uploaded_document` parameter. Return a 404 error if the reference doesn't already exist. 

Update the POST and PUT methods to return an error ([APPLICATIONS_CLOSED](https://github.com/deep-learning-indaba/Baobab/blob/bf99909f71ef7cca4923d3a93ddcea9e6d443ccf/api/app/utils/errors.py#L71)) if the `is_applications_open` flag on the relevant event is `False`

### PR DESCRIPTION
**Pull request by OdendaalG**
Fixes #592

### CODE REVIEW

File: api/app/references/api.py [hunk 1]
```diff
@@ -130,12 +131,37 @@ def post(self):
...
+    def put(self):
+        args = self.put_req_parser.parse_args()
+        token = args['token']
+        uploaded_document = args['uploaded_document']
+        reference_request = reference_request_repository.get_by_token(token)
+        event = event_repository.get_event_by_response_id(reference_request.response_id)
+        reference = reference_request_repository.get_reference_by_reference_request_id(reference_request.id)
+
+        if not reference_request:
+            return {}, 404
```
Comment (amritpurshotam) [inline api/app/references/api.py]: Perhaps in the api\app\utils\errors.py file add a new error `REFERENCE_REQUEST_NOT_FOUND = ({'message': 'No response found for the given event and user'}, 404)` and then import that and use it here

File: api/app/references/tests.py [hunk 2]
```diff
@@ -94,3 +95,66 @@ def test_reference_api(self):
...
+
+        response = self.app.get(
+            '/api/v1/reference', data={'response_id':1}, headers=self.headers)
+        LOGGER.debug(response.data)
+        data = json.loads(response.data)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(data), 1)
+        reference_request = reference_request_repository.get_by_id(1)
+        self.assertEqual(reference_request.has_reference(), True)
```
Comment (amritpurshotam) [inline api/app/references/tests.py]: Can we perhaps do an assert to see if the `uploaded_document` field actually changed to `DOCT-UPLOAD-79000`?

File: api/app/references/api.py (hunk 1 above)
Comment (OdendaalG) [reply api/app/references/api.py]: Added new error and used it in the reference API.

**Comment by OdendaalG** [comment]
Hi @amritpurshotam , I implemented the changes that you mentioned, but the issue is still open. I don't know if more changes have been requested, please let me know if there is something more I can do.
Regards

**Comment by OdendaalG** [comment]
No problem on the delay! I thought I had messed something up with the git process, thanks for the reply!

**Pull request merged**